### PR TITLE
Unvendor cpuinfo

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Because the `cpuinfo` module is required by `setup.py`, we need to pull it with `pyproject.toml`.

~~I think it is not needed in `requirements_dev.txt`.~~

TODO:

- [ ] Unit tests and/or doctests in docstrings
- [ ] `tox -e py310` passes locally
- [ ] Docstrings and API docs for any new/modified user-facing classes and functions
- [x] Changes documented in docs/release.rst
- [ ] `tox -e docs` passes locally
- [x] GitHub Actions CI passes
- [x] Test coverage to 100% (Coveralls passes)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [build-system]
-requires = ["setuptools>=40.8.0", "wheel", "Cython"]
+requires = ["setuptools>=40.8.0", "wheel", "Cython", "cpuinfo"]
```
Comment (jakirkham) [inline pyproject.toml:2]:
```suggestion
requires = ["setuptools>=40.8.0", "wheel", "Cython", "py-cpuinfo"]
```

File: requirements_rtfd.txt [hunk 2]
```diff
@@ -9,3 +9,4 @@ cython
 entrypoints
 zfpy==0.5.5 ; python_version < "3.11"
 typing-extensions
+cpuinfo==9.0.0
```
Comment (jakirkham) [inline requirements_rtfd.txt:12]:
```suggestion
py-cpuinfo==9.0.0
```

**Review by jakirkham — COMMENTED** [review]
On PyPI this is called [`py-cpuinfo`]( https://pypi.org/project/py-cpuinfo/ ), so suggesting updating requirements below accordingly

**Comment by jakirkham** [comment]
Thanks Dimitri! 🙏

**Review by jakirkham — APPROVED** [review]
Looks good. Thanks Dimitri! 🙏

**Comment by joshmoore** [comment]
Thanking of the upcoming release, best to bump to `0.11` with the new dependency?

**Comment by jakirkham** [comment]
Agreed. Opened issue ( https://github.com/zarr-developers/numcodecs/issues/377 ) so we can sort out anything for the next release. Hoping to have a couple more build fixes done, but will see. Depends on time constraints & help from others (like with `pyproject.toml`)

**Pull request merged**